ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -5452,11 +5454,17 @@ NtFreeVirtualMemory(IN HANDLE ProcessHandle,
                     // pages to decommit, and then push the VAD's starting address
                     // a bit further down, then decrement the commit charge
                     //
-                    // NOT YET IMPLEMENTED IN ARM3.
-                    //
-                    DPRINT1("Case A not handled\n");
-                    Status = STATUS_FREE_VM_NOT_AT_BASE;
-                    goto FailPath;
+                    MiLockProcessWorkingSetUnsafe(Process, CurrentThread);
+                    CommitReduction = MiCalculatePageCommitment(StartingAddress,
+                                                                EndingAddress,
+                                                                Vad,
+                                                                Process);
+                    Vad->u.VadFlags.CommitCharge -= CommitReduction;
+                    // For ReactOS: shrink the corresponding memory area
```

Comment:
Why "for ROS"? Does this suppose that Windows does more things there?

Reply:
`MEMORY_AREA` is a ROS-exclusive old Mm structure. We need to maintain them for memory regions owned by ARM3 so that old Mm doesn't think that the corresponding memory range is free and puts other stuff there.